documentation/design/software_design.md

```diff
@@ -0,0 +1,39 @@
+# **Gitops Plugin Design**
+Gitops software will provide ability to monitor git repositories for commits , changes etc and send the data to observability platforms like snappyflow.
+Gitops software will be provided as a standalone binary for different operating system. It will be supporting all major operating systems including linux(major distributions) , windows etc
+
+## **High level diagram**
+![HLD](./software_design_high_level.png)
```

Comment:
Create a "images" folder under "design" and move images there

Reply:
done